tx_service.go

```diff
@@ -3,53 +3,74 @@ package bux
 import (
 	"context"
 	"encoding/hex"
-	"errors"
+	"fmt"
 
-	"github.com/BuxOrg/bux/chainstate"
 	"github.com/BuxOrg/bux/utils"
-	"github.com/mrz1836/go-datastore"
 )
 
-// processTransactions will process transaction records
-func processTransactions(ctx context.Context, maxTransactions int, opts ...ModelOps) error {
-	queryParams := &datastore.QueryParams{
-		Page:          1,
-		PageSize:      maxTransactions,
-		OrderByField:  "created_at",
-		SortDirection: "asc",
-	}
+// The transaction is treat as external incoming transaction - transaction without a draft
+// Only use this function when you know what you are doing!
+func saveRawTransaction(ctx context.Context, c ClientInterface, allowUnknown bool, txHex string, opts ...ModelOps) (*Transaction, error) {
+	newOpts := c.DefaultModelOptions(append(opts, New())...)
+	tx := newTransaction(txHex, newOpts...)
 
-	conditions := map[string]interface{}{
-		"$or": []map[string]interface{}{{
-			blockHeightField: 0,
-		}, {
-			blockHeightField: nil,
-		}},
+	// Ensure that we have a transaction id (created from the txHex)
```

Comment:
Why? Is it possible to not have this ID? maybe this check should be in a `newTransaction` instead, and newTransaction should be able to return error? For me it looks like after `newTransaction` I still can have inconsistent state of the transaction. 

(This change could be made in separate PR later (or even cover by separated JIRA task) because it probably will force to change many places in code 🙂  )

Reply:
You're 100% right, it is like this because same function is used to init struct before reading tx from DB.

I'll in what impact this change will have and add new task in JIRA or make change here

Reply:
Okay, it demands changes in 12 files (a lot fo in tests). I'm going to push it in separate PR